sphinx.go

```diff
@@ -170,28 +192,31 @@ func (s *SphinxClient) Query(q *SphinxQuery) (*SphinxResult, error) {
 
 	conn, err := s.ConnectionPool.Get()
 	if err != nil {
-		// Type assertion as pool connection - have to since what is returned is
-		// base interface type.
-		if poolConn, ok := conn.(*pool.PoolConn); ok {
-			poolConn.MarkUnusable()
-		}
-		return nil, err
+		log.Printf("Error while getting connection from pool: %v", err)
+		s.RemoveBadConnection(conn)
+		return s.Query(q)
 	}
 	defer conn.Close()
 
 	_, err = headerBuf.WriteTo(conn)
 	if err != nil {
-		return nil, err
+		log.Printf("Error while writing to headerBuf: %v", err)
+		s.RemoveBadConnection(conn)
+		return s.Query(q)
```

Comment:
We should probably put in some sort of max # of retries - otherwise if it keeps failing this will just keep trying to run `Query` over and over.